great_expectations/experimental/datasources/postgres_datasource.py

```diff
@@ -115,7 +117,7 @@ class TableAsset(DataAsset):
     # Instance fields
     type: Literal["table"] = "table"
     table_name: str
-    column_splitter: Optional[ColumnSplitter] = None
+    column_splitter: Optional[SqlYearMonthSplitter] = None
```

Comment:
Why is this specific splitter expected? Won't there be more `ColumnSplitter`s to choose from?

Reply:
Yes, but this is the only one that exists now.
As we add more this will become a union of all the possible concrete splitters.

Reply:
Will there be no contributor `ColumnSplitter`s (will the universe of `ColumnSplitter`s be finite for a `Union`)?

Reply:
This is a good point @NathanFarmer. Likely splitters will be a point of extensibility.

Reply:
@NathanFarmer I'm not sure I understand. How does having a `Union` of splitter types preclude contributor-provided splitter types?

Is the point that this wouldn't work for `ColumnSplitter` plugins? That's true.

Reply:
We discovered that the existing splitters ARE NOT a point of extensibility and are restricted to these enums. As such we adding a custom validator to ensure that provided `method_name` matches this enum.

[great_expectations/great_expectations/execution_engine/split_and_sample/data_splitter.py](https://github.com/great-expectations/great_expectations/blob/c0aa33feedb23671d75f3f15408171c8ee417c3b/great_expectations/execution_engine/split_and_sample/data_splitter.py#L47-L60)